### PR TITLE
Replace XLA_COUNTER with TORCH_LAZY_COUNTER

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request replaces most of XLA_COUNTER with the pytorch/xla lib with upstream TORCH_LAZY_COUNTER. However, XLA_COUNTER is still kept to be used in the xla_client lib.

Test Plan:
python test/test_metrics.py MetricsTest.test_metrics_report
and CI.

### CODE REVIEW

**Review by JackCaoG — APPROVED** [review]
Thanks!

**Comment by alanwaketan** [comment]
@JackCaoG Will the REMOVE_TORCH_PIN label remove the pin before merging?

**Comment by JackCaoG** [comment]
lol nope, it is just a reminder for yourself. you can delete the pin and merge without waiting for CI

**Pull request merged**
